crates/binstalk/src/bins.rs

```diff
@@ -41,7 +42,22 @@ impl BinFile {
 
         // Generate install paths
         // Source path is the download dir + the generated binary path
-        let source_file_path = ctx.render(&data.meta.bin_dir)?;
+        let source_file_path = if let Some(bin_dir) = &data.meta.bin_dir {
+            ctx.render(bin_dir)?
+        } else {
+            let default_dirs = hosting::FULL_FILENAMES
+                .into_iter()
+                .chain(hosting::NOVERSION_FILENAMES)
+                .filter_map(|f| f.strip_suffix(".{ archive-format }"))
+                .chain([""])
+                .map(Path::new)
+                .map(|p| p.join("{ bin }{ binary-ext }"))
+            ;
+            // "{ bin }{ binary-ext }"
```

Comment:
Maybe we can try { bin }{ binary-ext } and bin/{ bin }{ binary-ext } here, just in case the tarball is packed more formal